src/common/main/kotlin/tech/thatgravyboat/skyblockapi/impl/events/TabListEventHandler.kt

```diff
@@ -152,4 +211,61 @@ object TabListEventHandler {
             this.footer = event.packet.footer()
         }
     }
+
+    private var logWidgets: Boolean = false
+    private val loggedWidgets = enumSetOf<TabWidget>()
+
+    @Subscription
+    fun onTabWidgetChange(event: TabWidgetChangeEvent) {
+        if (!logWidgets) return
+        if (event.widget !in loggedWidgets) return
+        Text.sendDebug("Tab widget changed: ") {
+            append(event.widget.name, TextColor.GOLD)
+            this.hover = Text.join(event.newComponents.toTypedArray(), CommonText.NEWLINE)
+            onClick {
+                McClient.clipboard = event.new.joinToString("\n")
+                Text.sendDebug("Copied tab widget contents to clipboard.")
+            }
+        }
+    }
+
+    @Subscription
+    fun onRegisterCommands(event: RegisterCommandsEvent) {
+        event.register("sbapi logtabwidgets") {
+            callback {
+                logWidgets = !logWidgets
+                Text.sendDebug("Tab widget logging is now") {
+                    if (logWidgets) append(" Enabled", TextColor.GREEN)
+                    else append(" Disabled", TextColor.RED)
+                }
+            }
+            thenCallback("list") {
+                if (loggedWidgets.isEmpty()) {
+                    Text.sendDebug("No tab widgets are being logged.")
+                } else {
+                    Text.sendDebug("Logged tab widgets: ${loggedWidgets.joinToString { it.name }}")
+                }
+            }
+            thenCallback("add widget", EnumArgument<TabWidget>()) {
+                val widget = argument<TabWidget>("widget")!!
+                if (loggedWidgets.add(widget)) {
+                    Text.sendDebug("Added ${widget.name} to logged tab widgets.")
+                } else {
+                    Text.sendDebug("${widget.name} is already being logged.")
+                }
+            }
+            thenCallback("remove widget", EnumArgument<TabWidget>()) {
+                val widget = argument<TabWidget>("widget")!!
+                if (loggedWidgets.remove(widget)) {
+                    Text.sendDebug("Removed ${widget.name} from logged tab widgets.")
+                } else {
+                    Text.sendDebug("${widget.name} is not being logged.")
+                }
+            }
+            thenCallback("clear") {
```

Comment:
```suggestion
            thenCallback("get widget", EnumArgument<TabWidget>()) {
                val widget = argument<TabWidget>("widget")!!
                widgets[widget]?.let {
                    Text.sendDebug("Contents of tab widget ${widget.name}") {
                        this.hover = Text.multiline(it)
                        onClick {
                            McClient.clipboard = it.joinToString("\n")
                            Text.sendDebug("Copied contents to clipboard.")
                        }
                    }
                } ?: Text.sendDebug("Tab widget ${widget.name} not present.")
            }
            thenCallback("clear") {
```

surely we can get this to test if it works 🙏